music_assistant/server/providers/tidal/__init__.py

```diff
@@ -350,24 +352,31 @@ async def get_playlist_tracks(
         tidal_session = await self._get_tidal_session()
         total_playlist_tracks = 0
         track: TidalTrack  # satisfy the type checker
-        async for track_obj in self._iter_items(
-            get_playlist_tracks, tidal_session, prov_playlist_id, limit=DEFAULT_LIMIT
-        ):
-            total_playlist_tracks += 1
-            track = self._parse_track(
-                track_obj=track_obj,
-                extra_init_kwargs={"position": total_playlist_tracks},
-            )
-            yield track
+        async with self._throttler:
+            with suppress(tidal_exceptions.TooManyRequests):
+                async for track_obj in self._iter_items(
+                    get_playlist_tracks,
+                    tidal_session,
+                    prov_playlist_id,
+                    limit=DEFAULT_LIMIT,
+                ):
+                    total_playlist_tracks += 1
+                    track = self._parse_track(
+                        track_obj=track_obj,
+                        extra_init_kwargs={"position": total_playlist_tracks},
+                    )
+                    yield track
 
     async def get_similar_tracks(self, prov_track_id: str, limit: int = 25) -> list[Track]:
         """Get similar tracks for given track id."""
         tidal_session = await self._get_tidal_session()
         async with self._throttler:
-            return [
-                self._parse_track(track)
-                for track in await get_similar_tracks(tidal_session, prov_track_id, limit)
-            ]
+            with suppress(tidal_exceptions.TooManyRequests):
```

Comment:
recurring theme, do not suppress like this
apply the backoff/retry logic to  get_similar_tracks and store that into a variable or use a walrus such as:

if tracks :=  get_similar_tracks(blah):
    return [parsetrack list comprehension]
return []